### PR TITLE
Apply redesign to Jetpack Search Congrats page

### PR DESCRIPTION
**Pull request by wongasy**
Related to 5892-gh-Automattic/dotcom-forge

## Proposed Changes

Apply redesign to the Jetpack Search Congrats Page.

|Before|After|
|------|------|
|Screen Shot 2024-03-06 at 3 29 42 PM|Screen Shot 2024-03-07 at 8 06 36 PM|

## Testing Instructions

- Purchase Jetpack Search on a Jetpack connected site, like a site with an Entrepreneur plan. If you already have it on a site, you can navigate to the respective Congrats Page with the following path:
```
/checkout/thank-you/:site/:receipt_id
```
- On that Congrats page, confirm you see the new design with the copy shown in the image above
- Test the main **Go to Search Dashboard** CTA, and confirm that it fires a `calypso_jetpack_product_thankyou` event with the following event props:
   - product_name: search
   - value: Customizer
   - site: wpcom
- Confirm that the CTA loaded `wp-admin/admin.php?page=jetpack-search`, which is in the Jetpack dashboard with some stats and settings, including a button to customize search results
- Purchase Jetpack on a non-Jetpack connected site, like a normal free site with now plan added to it yet.
- Confirm that the main CTA in this flow is a **Customize Search** button, instead of **Go to Search Dashboard**, and that it fires the same event as described above.
- Confirm that in this flow, the CTA loads `wp-admin/admin.php?page=jetpack-search`, which goes directly to the UI for customizing search results (skipping the Jetpack dashboard screen from the previous flow)
- Test the **Explore support resources** footer link, and confirm that it fires a `calypso_thank_you_footer_link_click` event with the following event props:
   - context: jetpack-search
   - type: generic-support
- Test the **All-in-one plugin documentation** footer link, and confirm that it fires a `calypso_thank_you_footer_link_click` event with the following event props:
   - context: jetpack-support
   - type: plugins-support

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/my-sites/checkout/checkout-thank-you/header.jsx [hunk 1]
```diff
@@ -413,18 +413,6 @@ export class CheckoutThankYouHeader extends PureComponent {
 		);
 	}
 
-	getSearchButtonProps() {
```
Comment (wongasy) [inline client/my-sites/checkout/checkout-thank-you/header.jsx:416]: Removing functions used for rendering the old congrats page.

File: client/my-sites/checkout/checkout-thank-you/redesign-v2/utils.ts [hunk 2]
```diff
@@ -47,7 +48,21 @@ export const isRefactoredForThankYouV2 = ( props: CheckoutThankYouCombinedProps
...
+			return true;
+		}
+
+		if ( isTitanMail( purchase ) ) {
+			return true;
+		}
+
+		if ( isJetpackSearchSlug( purchase.productSlug ) ) {
+			return true;
+		}
```
Comment (wongasy) [inline client/my-sites/checkout/checkout-thank-you/redesign-v2/utils.ts]: The `isRefactoredForThankYouV2()` function is used to determine whether or not to render the redesigned thank you page. We are adjusting the function here to flip over to the redesigned congrats page for Jetpack Search.

File: client/my-sites/checkout/checkout-thank-you/redesign-v2/pages/jetpack-search.tsx [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+import ThankYouProduct from 'calypso/components/thank-you-v2/product';
+import { recordTracksEvent } from 'calypso/lib/analytics/tracks';
+import { useSelector } from 'calypso/state';
+import {
+	getJetpackSearchCustomizeUrl,
+	getJetpackSearchDashboardUrl,
+} from 'calypso/state/sites/selectors';
+import { getSelectedSiteId, getSelectedSite } from 'calypso/state/ui/selectors';
+
+export default function JetpackSearchThankYou() {
```
Comment (wongasy) [inline client/my-sites/checkout/checkout-thank-you/redesign-v2/pages/jetpack-search.tsx]: `<JetpackSearchThankYou>` uses the redesigned component `<ThankYouV2>` to render the Congrats page for Jetpack Search. I basically took the copy and logic from `checkout/checkout-thank-you/header.jsx`, plugged them into `<ThankYouV2>`.

**Comment by chad1008** [comment]
Thanks again for getting the ball rolling on this one @wongasy. I think the [requirements outlined on the related issue](5892-gh-Automattic/dotcom-forge#issuecomment-1981748915) are addressed, looking forward to the first round of feedback.

File: client/my-sites/checkout/checkout-thank-you/redesign-v2/test/is-refactored-for-thank-you-v2.js [hunk 4]
```diff
@@ -88,4 +88,16 @@ describe( 'isRefactoredForThankYouV2', () => {
 			expect( isRefactoredForThankYouV2( props ) ).toBe( true );
 		}
 	} );
+
+	it( 'should return true for Jetpack Search', () => {
+		const props = {
+			receipt: {
+				data: {
+					purchases: [ { productSlug: 'jetpack_search' } ],
```
Comment (chad1008) [inline client/my-sites/checkout/checkout-thank-you/redesign-v2/test/is-refactored-for-thank-you-v2.js]: Is targeting a specific slug for this test too fragile? Would it be preferable to loop through each of the [Jetpack Search product slugs](https://github.com/Automattic/wp-calypso/commit/77adc2c4bffa8059f47d9daad12a856a2b61a8e5)?

Comment (wongasy) [reply client/my-sites/checkout/checkout-thank-you/redesign-v2/test/is-refactored-for-thank-you-v2.js]: Good question! This made me realize that I should have used `isSearch()` (function initially defined in `header.jsx` to check instead of `isJetpackSearchProductSlug()`. That function checks against `productType` instead of `productSlug`.

https://github.com/Automattic/wp-calypso/pull/88262/files#diff-5836f17692fa28c8c636e8d40f35daac248a6ee35e0f71130d05f9300605e1efL65

File: client/my-sites/checkout/checkout-thank-you/redesign-v2/pages/jetpack-search.tsx [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+			buttonHref: localizeUrl( 'https://wordpress.com/support/' ),
+			buttonOnClick: () => {
+				recordTracksEvent( 'calypso_thank_you_footer_link_click', {
+					context: 'jetpack-search',
+					type: 'generic-support',
+				} );
+			},
+		},
+		{
+			key: 'footer-plugins-support',
```
Comment (wongasy) [inline client/my-sites/checkout/checkout-thank-you/redesign-v2/pages/jetpack-search.tsx:31]: @chad1008 A heads up that I also added a second footer blurb (copy taken from the marketplace Congrats page: https://github.com/Automattic/wp-calypso/blob/b60cadaf8602ef05ea2f39d6712baa6a840d5bab/client/my-sites/checkout/checkout-thank-you/marketplace/use-thank-you-footer.tsx#L72-L78), thought it'd be appropriate to also provide plugin information, given that the user just installed the Jetpack Search plugin.

File: client/my-sites/checkout/checkout-thank-you/redesign-v2/products/jetpack-search-product.tsx [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+		recordTracksEvent( 'calypso_jetpack_product_thankyou', {
+			product_name: 'search',
+			value: 'Customizer',
+			site: 'wpcom',
+		} );
+	};
+
+	return (
+		<ThankYouProduct
+			name={ purchase.productName }
```
Comment (wongasy) [inline client/my-sites/checkout/checkout-thank-you/redesign-v2/products/jetpack-search-product.tsx:49]: @chad1008 Note that I also updated the code such that we get the product name from the `purchase` object instead of hardcoding the product name "Jetpack Search".

File: client/my-sites/checkout/checkout-thank-you/redesign-v2/products/jetpack-search-product.tsx [hunk 7]
```diff
@@ -28,10 +28,10 @@ export default function ThankYouJetpackSearchProduct( {
 		getJetpackSearchDashboardUrl( state, siteId as number )
 	);
 
-	const productButtonLabel = selectedSite?.jetpack
```
Comment (chad1008) [inline client/my-sites/checkout/checkout-thank-you/redesign-v2/products/jetpack-search-product.tsx]: @wongasy I definitely agree that the site should always be defined, but TypeScript sees `selectedSite` as potentially `null` or `undefined` and throws an error here. I opted for optional chains to suppress that, but we could also go another route if you prefer? Maybe an `if ( selectedSite ) ...` check? That would lead to both variables being `undefined` if the site somehow wasn't truthy, but that shouldn't happen at this point anyway, I don't think. Curious what your thoughts on best approach 🙂

Comment (chad1008) [reply client/my-sites/checkout/checkout-thank-you/redesign-v2/products/jetpack-search-product.tsx]: Playing around with this a bit more, I think I'm still leaning towards the optional chaining to resolve the type error.

Thought process:
- both of these variables (`productButtonLabel` and `productButtonHref`) are being declared via a ternaries based on the truthiness of `siteId.jetpack`.
- with optional chaining, if `siteId` is somehow `null` or `undefined`, the ternary will evaluate as `undefined` and return a falsey result... which is exactly what we'd want to happen, since in this case `siteId.jetpack` is also automatically `undefined` and falsey. 
- Basically if `siteId` proves to be falsey, the behavior of this code will be the same as if the `jetpack` prop itself were falsey, which is actually what we're trying to evaluate in the first place.

I'm hoping my description here makes sense 😅 If it does, perhaps I just add a quick comment explaining the reasoning behind the optional chaining?

If not, happy to explore/learn a better way to tackle this one!

Comment (wongasy) [reply client/my-sites/checkout/checkout-thank-you/redesign-v2/products/jetpack-search-product.tsx]: Thanks for the explanation! I didn't realize the optional was meant to resolve the type error. I'll revert back to conditional chaining.

File: client/my-sites/checkout/checkout-thank-you/redesign-v2/pages/jetpack-search.tsx [hunk 8]
```diff
@@ -5,16 +5,13 @@ import { recordTracksEvent } from 'calypso/lib/analytics/tracks';
...
-import type { ThankYouJetpackSearchProductProps } from '../products/jetpack-search-product';
 import type { ReceiptPurchase } from 'calypso/state/receipts/types';
 
 export type JetpackSearchThankYouProps = {
 	purchase: ReceiptPurchase;
 };
 
-export default function JetpackSearchThankYou( {
-	purchase,
-}: Omit< ThankYouJetpackSearchProductProps, 'siteId' > ) {
```
Comment (wongasy) [inline client/my-sites/checkout/checkout-thank-you/redesign-v2/pages/jetpack-search.tsx]: @chad1008 I noticed you made this change https://github.com/Automattic/wp-calypso/pull/88262/commits/b6275c8ebca4741b1e3dfa53d456bf52ddcf1ff0 to resolve TS error. The initial problem was actually the fact that I supplied the wrong type 😓 It should be fixed by https://github.com/Automattic/wp-calypso/pull/88262/commits/9b931a2b201e0a874be1e6dcaca8584de0c7670b

Comment (chad1008) [reply client/my-sites/checkout/checkout-thank-you/redesign-v2/pages/jetpack-search.tsx]: Ah, cool. The updated types look good to me 😄

**Comment by chad1008** [comment]
@wongasy Added one small change to the new footer link event props. Both footer links now carry `context: jetpack-search` to denote what kind of transaction they came from, with the new event carrying `type: plugin-support` to differentiate exactly what this link points to.

Testing instructions added, and we should be good to go!

File: client/my-sites/checkout/checkout-thank-you/redesign-v2/products/jetpack-search-product.tsx [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+	const productButtonHref = selectedSite?.jetpack
+		? jetpackSearchDashboardUrl
+		: jetpackSearchCustomizeUrl;
+
+	const recordThankYouClick = () => {
+		recordTracksEvent( 'calypso_jetpack_product_thankyou', {
+			product_name: 'search',
+			value: 'Customizer',
+			site: 'wpcom',
+		} );
```
Comment (wongasy) [inline client/my-sites/checkout/checkout-thank-you/redesign-v2/products/jetpack-search-product.tsx]: @chad1008 I came to realize that there's probably a bug in the current tracking code, as we're reporting the same value regardless of which button we render ("Go to Search Dashboard" or "Customize Search"). Let's take the opportunity to fix tracking such that we report different values for a Jetpack site:
* `value: 'Search Dashboard'`
* `site: 'jetpack'`

**Comment by wongasy** [comment]
Thanks @chad1008 for pumping out the test instructions! I found one last feedback that we should address: https://github.com/Automattic/wp-calypso/pull/88262/files/a51298b3c4ef295056de2a8244131996a065c472#r1518292387

We can do one last round of testing on Monday once the feedback is addressed, then the PR should be good to go!

File: client/my-sites/checkout/checkout-thank-you/redesign-v2/products/jetpack-search-product.tsx (hunk 9 above)
Comment (chad1008) [reply client/my-sites/checkout/checkout-thank-you/redesign-v2/products/jetpack-search-product.tsx]: Done! 773fe93134

The two CTA flavors will now track separately. Glad you spotted that!

File: client/my-sites/checkout/checkout-thank-you/redesign-v2/products/jetpack-search-product.tsx [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+	const productButtonHref = selectedSite?.jetpack
+		? jetpackSearchDashboardUrl
+		: jetpackSearchCustomizeUrl;
+	const thankYouEventProps = selectedSite?.jetpack
+		? { product_name: 'search', value: 'Search Dashboard', site: 'jetpack' }
+		: { product_name: 'search', value: 'Customizer', site: 'wpcom' };
+
+	const recordThankYouClick = () => {
+		recordTracksEvent( 'calypso_jetpack_product_thankyou', thankYouEventProps );
+	};
```
Comment (wongasy) [inline client/my-sites/checkout/checkout-thank-you/redesign-v2/products/jetpack-search-product.tsx]:
```suggestion

	const recordThankYouClick = () => {
		recordTracksEvent( 'calypso_jetpack_product_thankyou', {
			product_name: 'search',
			value: selectedSite?.jetpack ? 'Search Dashboard' : 'Customizer',
			site: selectedSite?.jetpack ? 'jetpack' : 'wpcom',
		} );
	};
```

What if we conditionally set the `value` and `site` event properties? That way, we avoid conditionally defining the entire object.

**Review by wongasy — COMMENTED** [review]
LGTM and tests well! If we address the final feedback around the tracking event prop. We should be good!

* "Go to Search Dashboard" CTA is rendered for Jetpack site.

Screen Shot 2024-03-08 at 7 29 05 PM

* "Go to Search Dashboard" CTA takes users to the search dashboard and the correct click tracking event is fired.

Screen Shot 2024-03-08 at 7 29 29 PM

* "Customize Search" CTA is rendered for wpcom site.

Screen Shot 2024-03-08 at 7 29 58 PM

* "Customize Search" CTA takes users to the search dashboard and the correct click tracking event is fired.

Screen Shot 2024-03-08 at 7 30 10 PM

* On click of "Explore support resources" link on the congrats page trigger the correct tracking event.

Screen Shot 2024-03-08 at 7 30 29 PM

* On click of "Plugin documentation" link on the congrats page trigger the correct tracking event.

Screen Shot 2024-03-08 at 7 30 34 PM
